front/src/services/helpers/formatSearchParamToPrismaQuery.ts

```diff
@@ -1,41 +1,50 @@
-import { Order, TimeOption } from "@/types/logDisplayOptions";
+import { LogDisplayOptions, Order, SortOptions, TimeOption } from "@/types/logDisplayOptions";
 import { llm_logs } from "@prisma/client";
 import {
     PrismaSort,
     SearchCondition,
     Timeframe,
 } from "../types/queryConditions";
+import { MS_PER_DAY, MS_PER_HOUR, MS_PER_WEEK } from "./timeConstants";
 
-const MS_PER_HOUR = 60 * 60 * 1000;
-const MS_PER_DAY = 24 * MS_PER_HOUR;
-const MS_PER_WEEK = 7 * MS_PER_DAY;
+export const convertSearchParamToPrismaConditions = (searchParams?: URLSearchParams) => {
+    const data = extractDataFromSearchParams(searchParams);
 
-export const extractDataFromSearchParam = (searchParams?: URLSearchParams) => {
-    const params = new URLSearchParams(searchParams);
-
-    const selectedTime =
-        (params.get("dateTimeFilter") as TimeOption) || "Last hour";
-    const timeframe = stringToTimeframeObject(selectedTime);
+    let timeframe: Timeframe;
+    if (data.dateTimeFilter === "Custom interval" && data.startDateTime && data.endDateTime)
```

Comment:
(tech steps) if implementing the previously mentioned steps, then this needs to check if dateTimeFilter is undefined

Reply:
It depends on how we want to handle the priority. I just removed the first condition. The new behaviour is that if the boundaries are defined, then they are used to create the timeframe, but if at least one of them is not, then stringToTimeframeObject is called.